File: rides-service/src/main/java/com/example/ride/client/DriverServiceClientFallback.java
package com.example.ride.client;

import com.example.ride.util.ExceptionMessages;
import org.springframework.http.ResponseEntity;
import org.springframework.stereotype.Component;

@Component
public class DriverServiceClientFallback implements DriverServiceClient {
    @Override
    public ResponseEntity<Boolean> doesDriverExists(Long driverId) {
        throw new RuntimeException(ExceptionMessages.UNABLE_TO_REACH_DRIVER_SERVICE.format());
    }
}

File: rating-service/src/main/kotlin/com/example/rating_service/client/DriverServiceClientFallback.kt
package com.example.rating_service.client

import com.example.rating_service.util.ExceptionMessages
import org.springframework.http.ResponseEntity
import org.springframework.stereotype.Component

@Component
class DriverServiceClientFallback: DriverServiceClient {
    override fun doesDriverExist(driverId: Long): ResponseEntity<Boolean> {
        throw RuntimeException(ExceptionMessages.UNABLE_TO_REACH_DRIVER_SERVICE)
    }
}